### ISSUE TITLE
'ImportDeclaration': 'never', is not enforced.

### ISSUE THREAD

**Issue by TheDutchCoder** [eslint/eslint#14024]
**Tell us about your environment**

* **ESLint Version:** 6.7.2
* **Node Version:** 12.19.0
* **npm Version:** 6.14.10

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  root: true,
  env: {
    node: true,
  },
  extends: [
    'plugin:vue/vue3-recommended',
    '@vue/typescript/recommended',
  ],
  parserOptions: {
    ecmaVersion: 2020,
  },
  ignorePatterns: ['**/models/*.*'],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
    '@typescript-eslint/indent': [
      'error',
      2,
    ],
    '@typescript-eslint/ban-ts-ignore': 0,
    'linebreak-style': [
      'error',
      'unix',
    ],
    'quotes': [
      'error',
      'single', {
        'allowTemplateLiterals': true,
      },
    ],
    'semi': [
      'error',
      'never',
    ],
    'space-before-function-paren': [
      'error',
      'always',
    ],
    'object-curly-spacing': [
      'error',
      'always',
    ],
    'object-curly-newline': ['error', {
      'ObjectExpression': 'always',
      'ObjectPattern': { 'multiline': true },
      'ImportDeclaration': 'never',
      'ExportDeclaration': { 'multiline': true, 'minProperties': 2 }
    }],
    'comma-dangle': [
      'error',
      {
        'arrays': 'always-multiline',
        'objects': 'always-multiline',
        'imports': 'always-multiline',
        'exports': 'always-multiline',
        'functions': 'never',
      },
    ],
  },
  overrides: [
    {
      files: [
        '**/__tests__/*.{j,t}s?(x)',
        '**/tests/unit/**/*.spec.{j,t}s?(x)',
      ],
      env: {
        jest: true,
      },
    },
  ],
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

This is in a .vue SFC, within the script tag.
```js
import { defineComponent, ref,
  PropType } from 'vue'
```

ESlint is ran from within VSCode.

**What did you expect to happen?**
The import to not allow line breaks, because `'ImportDeclaration': 'never',` is set.

**What actually happened? Please include the actual, raw output from ESLint.**
Line breaks aren't throwing an error.

**Are you willing to submit a pull request to fix this bug?**
I have no knowledge of eslint internals, but sure, if I can help, more than willing to!

**Comment by TheDutchCoder** [comment]
Please not that the docs are very explicit about what is supposed to happen here:
`"never" disallows line breaks inside braces`

And there's definitively a line break between these braces:
```js

import { defineComponent, ref, // looks like a line break to me!
  PropType } from 'vue'
```

**Comment by mdjermanovic** [comment]
Hi @TheDutchCoder, thanks for the issue!

This rule enforces linebreaks after `{` and before `}`.

In this case, it works as intended, since there's no linebreak between `{` and `defineComponent`, and there's no linebreak between `PropType` and `}`.

There are examples of correct code for this rule with the "never" option with linebreaks inside but not after `{` and before `}`:

```js
/*eslint object-curly-newline: ["error", "never"]*/

let d = {foo: 1,
    bar: 2};
let e = {foo: function() {
    dosomething();
}};
```

**Comment by TheDutchCoder** [comment]
Thanks for replying!

The rule clearly says "inside" braces, so maybe the output is intended, but then the description of the rule is incorrect.

Regardless, what would be the rule to make sure no line breaks are allowed, well, inside the braces? ;)

**Comment by mdjermanovic** [comment]
> The rule clearly says "inside" braces, so maybe the output is intended, but then the description of the rule is incorrect.

I agree that the wording isn't clear, a PR to clarify this would be welcome!

> Regardless, what would be the rule to make sure no line breaks are allowed, well, inside the braces? ;)

We don't have a rule to disallow linebreaks inside braces of named imports in the core set of rules. Given [our rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), I think the best course of action would be to create a custom rule or a plugin.

**Comment by arminyahya** [comment]
@TheDutchCoder did you start writing a custom rule for this? I'll be happy to try.
can i change object-curly-newline docs for this issue?

**Comment by mdjermanovic** [comment]
> can i change object-curly-newline docs for this issue?

Yes, I think it's a good idea to clarify what "inside braces" means for this rule, or to reword it.

**Comment by arminyahya** [comment]
@mdjermanovic 

Isn't better to replace **inside braces** with **for braces**?

sentences change like this:

`"always"` requires line break for braces.

**Comment by mdjermanovic** [comment]
I think it would be best to say "after" and "before", like in rule details of [array-bracket-newline](https://eslint.org/docs/rules/array-bracket-newline).

**Comment by arminyahya** [comment]
I'm going to do this, give me one day.
